maven-core/src/main/java/org/apache/maven/project/DefaultProjectBuilder.java

```diff
@@ -263,21 +264,14 @@ private DependencyResolutionResult resolveDependencies( MavenProject project, Re
 
     private List<String> getProfileIds( List<Profile> profiles )
     {
-        List<String> ids = new ArrayList<>( profiles.size() );
-
-        for ( Profile profile : profiles )
-        {
-            ids.add( profile.getId() );
-        }
-
-        return ids;
+        return profiles.stream().map( Profile::getId ).collect( Collectors.toList() );
     }
 
     private ModelBuildingRequest getModelBuildingRequest( InternalConfig config )
     {
         ProjectBuildingRequest configuration = config.request;
 
-        ModelBuildingRequest request = new DefaultModelBuildingRequest();
+        DefaultModelBuildingRequest request = new DefaultModelBuildingRequest();
```

Comment:
This change is necessary to pass the builder?

Reply:
Not anymore, during refactoring there was a period where I needed it. Changed it back.